### PR TITLE
[#3202] improvement(core): add name spec for catalog and metalake

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

 - Check name spec when creating and renaming catalogs or metalakes

### Why are the changes needed?

Fix: #3202 

### Does this PR introduce _any_ user-facing change?

yes, catalogs and metadata lakes have a much stricter name specification.

### How was this patch tested?

tests added

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogNormalizeDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+   * Regular expression explanation:
+   *
+   * <p>^[a-zA-Z_] - Starts with a letter, digit, or underscore
+   *
+   * <p>[a-zA-Z0-9_]{0,63} - Followed by 0 to 63 characters (making the total length at most 64) of
+   * letters (both cases), digits, underscores
+   *
+   * <p>$ - End of the string
+   */
+  private static final String CATALOG_NAME_PATTERN = "^\\w[\\w]{0,63}$";
```
[the remark this answers is not in the document]
Comment (mchades) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogNormalizeDispatcher.java:35]: fixed

File: core/src/test/java/com/datastrato/gravitino/catalog/TestCatalogNormalizeDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    catalogIdent2, RELATIONAL, "test", null, null));
+    Assertions.assertEquals("The catalog name 'a-b' is illegal.", exception.getMessage());
+
+    exception =
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () ->
+                catalogNormalizeDispatcher.alterCatalog(
+                    catalogIdent2, CatalogChange.rename("a-b")));
+    Assertions.assertEquals("The catalog name 'a-b' is illegal.", exception.getMessage());
```
[the remark this answers is not in the document]
Comment (mchades) [reply core/src/test/java/com/datastrato/gravitino/catalog/TestCatalogNormalizeDispatcher.java]: added

File: core/src/test/java/com/datastrato/gravitino/catalog/TestCatalogNormalizeDispatcher.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    if (catalogManager != null) {
+      catalogManager.close();
+      catalogManager = null;
+    }
+  }
+
+  @Test
+  public void testNameSpc() {
```
Comment (jerryshao) [inline core/src/test/java/com/datastrato/gravitino/catalog/TestCatalogNormalizeDispatcher.java:81]: `testNameSpec`.

Comment (mchades) [reply core/src/test/java/com/datastrato/gravitino/catalog/TestCatalogNormalizeDispatcher.java]: fixed

**Review by jerryshao — APPROVED** [review]

**Pull request merged**